### PR TITLE
Imran/lp 1369

### PR DESCRIPTION
**Pull request by emraan004**
add course open date to course pre experience module

### CODE REVIEW

File: common/djangoapps/custom_settings/views.py [hunk 1]
```diff
@@ -69,3 +77,31 @@ def course_custom_settings(request, course_key_string):
...
+    """
+    this method compares course start date and course open date
+    and make sure that course open date is greater than start date
+    :param settings:
+    :param course_open_date:
+    :return:
+        validated course open date or none
+    """
+
+    if course_open_date and course_open_date != '':
```
Comment (saad-waqar) [inline common/djangoapps/custom_settings/views.py]: ```if course_open_date``` would be sufficient let's change this

File: common/djangoapps/custom_settings/views.py [hunk 2]
```diff
@@ -69,3 +77,31 @@ def course_custom_settings(request, course_key_string):
 
         else:
             return HttpResponseNotAllowed("Bad Request", content_type="text/plain")
+
+
+def validate_course_open_date(settings, course_open_date):
```
Comment (saad-waqar) [inline common/djangoapps/custom_settings/views.py]: can we verify if any of the methods in common.djangoapps.util.date_utils might help here? (and any other place in this change)

File: lms/djangoapps/philu_overrides/helpers.py [hunk 3]
```diff
@@ -71,18 +71,30 @@ def get_course_next_classes(request, course):
     from student.models import CourseEnrollment
     from opaque_keys.edx.locations import SlashSeparatedCourseKey
     from course_action_state.models import CourseRerunState
+    from openedx.features.course_card.helpers import get_course_open_date
+
+    courses = []
```
Comment (saad-waqar) [inline lms/djangoapps/philu_overrides/helpers.py]: can we rename this to something like future_courses?

File: common/djangoapps/custom_settings/views.py (hunk 1 above)
Comment (emraan004) [reply common/djangoapps/custom_settings/views.py]: Sure

File: common/djangoapps/custom_settings/views.py (hunk 2 above)
Comment (emraan004) [reply common/djangoapps/custom_settings/views.py]: No there is not and this is specific to course cards and custom settings

File: lms/djangoapps/philu_overrides/helpers.py (hunk 3 above)
Comment (emraan004) [reply lms/djangoapps/philu_overrides/helpers.py]: This method just returns course open date and not any courses

File: common/djangoapps/custom_settings/views.py [hunk 4]
```diff
@@ -51,15 +55,19 @@ def course_custom_settings(request, course_key_string):
                     'show_grades': settings.show_grades,
                     'tags': settings.tags,
                     'seo_tags': "" if not settings.seo_tags else json.loads(settings.seo_tags),
+                    'course_open_date': ""
```
Comment (ahmed-faraz46) [inline common/djangoapps/custom_settings/views.py]: can we please implement a small helper for this?

File: common/static/css/tinymce-content-philu-overrides.css [hunk 5]
```diff
@@ -477,3 +477,14 @@ a:hover.tooltips span {
...
+.view-settings .group-settings.advanced-policies .course-advanced-policy-list-item .field input {
+    width: 100%;
+    box-sizing: border-box;
+    color: #cb9c40;
+    box-shadow: 0 1px 2px rgba(0,0,0,0.1) inset;
+    background-color: #edf1f5;
+    background-image: -webkit-linear-gradient(#edf1f5,#fdfefe);
+    background-image: linear-gradient(#edf1f5,#fdfefe);
+    border: 1px solid #b0b6c2;
+}
```
Comment (ahmed-faraz46) [inline common/static/css/tinymce-content-philu-overrides.css]: EOF please!

File: lms/djangoapps/philu_overrides/helpers.py (hunk 3 above)
Comment (ahmed-faraz46) [reply lms/djangoapps/philu_overrides/helpers.py]: @emraan004 how about to define a method to get course_runs as we defined `get_course_current_class` to get current classes.

**Pull request merged**